CMSgov/mint-app
# [EASI-4536] (Backend) Model Plan Anticipated Statuses

# EASI-4536

## Description


A functional resolver, AnticipatedStatuses, has been created on ModelPlan to determine anticipated statuses based on the following conditions.

- “Complete ICIP” date in anticipated timeline has passed but status is not "ICIP completed" or a future status
- “Clearance start date” date in anticipated timeline has passed but status is not any of the clearance statuses or a future status
- “Clearance end date” date in anticipated timeline has passed but status is not "Cleared" or a future status
- “Announce model” date in anticipated timeline has passed but status is not "Announced" or a future status
- “Performance start date” date in anticipated timeline has passed but status is not "Active" or a future status
- “Performance end date” date in anticipated timeline has passed but status is not "Ended"
- For all of the above, modal should only show if status is NOT "Paused" or "Canceled"

## How to test this change


- Create a model plan
- Set dates according to what you'd like to test in Plan Basics
- Use a Postman query to get the corresponding ModelPlan
- Assert that you are provided with the appropriate corresponding anticipated statuses

> [!NOTE]
> A date on the current day is considered to have passed. I could see reasoning going either way on this. I left the default behavior because it seems a bit more intuitively appropriate, besides being mathematically correct.


## PR Author Checklist




## PR Reviewer Guidelines

- It's best to pull the branch locally and test it, rather than just looking at the code online!
- When approving a PR, provide a reason _why_ you're approving it
  - e.g. "Approving because I tested it locally and all functionality works as expected"
  - e.g. "Approving because the change is simple and matches the Figma design"
- Don't be afraid to leave comments or ask questions, especially if you don't understand why something was done! (This is often a great time to suggest code comments or documentation updates)
- Check that all code is adequately covered by tests - if it isn't feel free to suggest the addition of tests.
